### PR TITLE
Improve `with-algolia-react-instantsearch` example and convert to TypeScript

### PR DESCRIPTION
**Pull request by maxproske**
Converted to TypeScript to match Contribution docs, and updated/simplified the example.

- Replaced stylesheets in Head component with imported styles
- Removed `style-loader`, `css-loader`, `cross-env`, `prop-types` packages
- Removed custom webpack config

## Documentation / Examples

- [X] Make sure the linting passes by running `pnpm build && pnpm lint`
- [X] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: examples/with-algolia-react-instantsearch/utils/index.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type { SearchState } from 'react-instantsearch-core'
+import qs from 'qs'
+
+export const createURL = (state: SearchState) => `?${qs.stringify(state)}`
+
+export const pathToSearchState = (path: string) =>
+  path.includes('?') ? qs.parse(path.substring(path.indexOf('?') + 1)) : {}
+
+export const searchStateToURL = (searchState: SearchState) =>
+  searchState ? `${window.location.pathname}?${qs.stringify(searchState)}` : ''
```
Comment (balazsorban44) [inline examples/with-algolia-react-instantsearch/utils/index.ts:10]: Could we just use `URL` and `URLSearchParams` instead and get rid of `qs`?

File: examples/with-algolia-react-instantsearch/package.json [hunk 2]
```diff
@@ -2,19 +2,25 @@
...
   "dependencies": {
-    "algoliasearch": "4.3.0",
-    "cross-env": "^7.0.2",
-    "css-loader": "1.0.0",
+    "algoliasearch": "^4.14.2",
+    "instantsearch.css": "^7.4.5",
     "next": "latest",
-    "prop-types": "^15.5.10",
-    "qs": "^6.4.0",
+    "qs": "^6.11.0",
```
Comment (balazsorban44) [inline examples/with-algolia-react-instantsearch/package.json:12]: let's try to get rid of this

**Review by balazsorban44 — CHANGES_REQUESTED** [review]
Thanks, only had a minor comment

File: examples/with-algolia-react-instantsearch/utils/index.ts (hunk 1 above)
Comment (maxproske) [reply examples/with-algolia-react-instantsearch/utils/index.ts:10]: @balazsorban44 I looked into it, and unfortunately `URLSearchParams` does not support nesting, which `react-instantsearch` search state does: https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/URLSearchParams#parameters

**Comment by maxproske** [comment]
Thank you @balazsorban44. I left a comment above!

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**